File: frontend/src/app/models/carro.ts
import {Marca} from './marca';
import {Proprietario} from './proprietario';

export class Carro {
  id! : number;
  nome!: string;
  marca!: Marca;
  proprietarios: Proprietario[] = [];

  constructor(id: number, nome: string, marca: Marca | null) {
    this.id = id;
    this.nome = nome;
    if (marca) this.marca = marca;
  }
}

File: backend/src/main/java/br/com/carros/carros/entity/Carro.java
package br.com.carros.carros.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Carro {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    private int anoLancamento;

    @ManyToOne
    @JsonIgnoreProperties("carros")
    private Marca marca;


    @ManyToMany
    @JoinTable(name = "carro_proprietario")
    private List<Proprietario> proprietarios;


}



